source/simulator/helper_functions.cc

```diff
@@ -1207,6 +1207,76 @@ namespace aspect
     compute_depth_average(values, f);
   }
 
+  namespace
+  {
+    template <int dim>
+    class FunctorDepthAverageHeatFlux
```

Comment:
Here (and maybe in other places), I would actually prefer a name that emphasizes the fact that you compute the *vertical* heat flux. To me, a flux is a vector quantity, but you only compute a scalar quantity -- but the names don't reflect this.